zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/SendAckRequestProcessor.java

```diff
@@ -46,7 +48,7 @@ public void processRequest(Request si) {
                 learner.writePacket(qp, false);
             } catch (IOException e) {
                 LOG.warn("Closing connection to leader, exception during packet send", e);
-                learner.closeSockSync();
+                learner.closeSocket();
```

Comment:
+1 on this.

This is the leftover of ZOOKEEPER-4409. We should respect `zookeeper.learner.closeSocketAsync` here.